src/SFML/Window/Android/JniHelper.cpp

```diff
@@ -242,39 +242,49 @@ std::optional<JniList<JniMotionRange, JniMotionRangeClass>> JniInputDevice::getM
 
 
 ////////////////////////////////////////////////////////////
-Jni::Jni(Jni&& other) noexcept
+Jni::Jni(JavaVM& vm, JNIEnv& env) : m_vm(&vm), m_env(&env)
 {
-    std::swap(m_vm, other.m_vm);
 }
 
 
 ////////////////////////////////////////////////////////////
-Jni::Jni(JavaVM* vm) : m_vm(vm)
+Jni::Jni(Jni&& other) noexcept
 {
+    std::swap(m_vm, other.m_vm);
+    std::swap(m_env, other.m_env);
 }
 
 
 ////////////////////////////////////////////////////////////
 Jni::~Jni()
 {
+    m_env = nullptr;
```

Comment:
There is no need to change the value of data members in a destructor. This is just dead code.